### ISSUE TITLE
There is no way to set the DisplayName for the OpenIdConnect provider

### ISSUE THREAD

**Issue by Shazwazza** [AzureAD/microsoft-identity-web#808]
**Which version of Microsoft Identity Web are you using?**
1.3.0

**Where is the issue?**
* Other (please describe)

It's not possible to set the display name since the call to AddOpenIdConnect is all done internally here https://github.com/AzureAD/microsoft-identity-web/blob/c75ea22dd4cf99ff6fc30c805d3d0cffcbc76879/src/Microsoft.Identity.Web/WebAppExtensions/MicrosoftIdentityWebAppAuthenticationBuilderExtensions.cs#L254 which doesn't pass in a display name and a display name is not configurable by AuthenticationSchemeOptions

**Is this a new or an existing app?**
c. This is a new app or an experiment.

**Expected behavior**
There should be a parameter in  `AddMicrosoftIdentityWebApp` to supply a DisplayName

**Actual behavior**
There is not an option to do this

**Comment by Shazwazza** [comment]
I'm happy to send a PR to add this

**Comment by jmprieur** [comment]
The display name (of the signed-in user, I'm assuming) is inferred from the claims which are in the IDToken. There is no need to specify it externally. What is your scenario? what are you trying to do?. In which case would you want to override a display name which is not in the claims?

To answer your question:

1. you have full control of your UI anyway. When you unfold one of the ASP.NET Core templates, the page displaying the name is [Pages/Shared/_LoginPartial](https://github.com/AzureAD/microsoft-identity-web/blob/c75ea22dd4cf99ff6fc30c805d3d0cffcbc76879/ProjectTemplates/templates/RazorPagesWeb-CSharp/Pages/Shared/_LoginPartial.OrgAuth.cshtml#L44), which is in the code of your app (not in the framework / the library).

2. if you don't want to change the UI, the display comes from the `options.TokenValidationParameters.NameClaimType` 

   https://github.com/AzureAD/microsoft-identity-web/blob/c75ea22dd4cf99ff6fc30c805d3d0cffcbc76879/src/Microsoft.Identity.Web/WebAppExtensions/MicrosoftIdentityWebAppAuthenticationBuilderExtensions.cs#L272-L279

   You can always override after calling AddMicrosoftIdentityWebApp

   ```CSharp
   services.Configure<OpenIdConnectOptions>(OpenIdConnectDefaults.AuthenticationScheme, options =>
   {
       options.TokenValidationParameters.NameClaimType = nameOfTheClaimYouWantToUse;
   });
   ```

**Comment by Shazwazza** [comment]
Hi there, what I am referring to by `DisplayName` is the display name of the provider.

The code is what I mentioned above with this call `builder.AddOpenIdConnect`. This method has an optional `displayName` parameter which must be set via this call as its not settable by any options configuration. [This is the overload I'm referring to](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.openidconnectextensions.addopenidconnect?view=aspnetcore-5.0#Microsoft_Extensions_DependencyInjection_OpenIdConnectExtensions_AddOpenIdConnect_Microsoft_AspNetCore_Authentication_AuthenticationBuilder_System_String_System_String_System_Action_Microsoft_AspNetCore_Authentication_OpenIdConnect_OpenIdConnectOptions__) which sets "A display name for the authentication handler."

**Comment by jmprieur** [comment]
Thanks for the update, @Shazwazza 
This makes sense.

**Comment by farshid3003** [comment]
I have this problem as well, we cannot set the provider display name

**Comment by jmprieur** [comment]
@Shazwazza @farshid3003 
would it work if we predefined a better name like "Microsoft identity platform" ? or "Azure active directory" ? or do you need to provide your own label? (what would it be?)

**Comment by Shazwazza** [comment]
@jmprieur Since this lib can be used as boilerplate configuration for Azure B2C then the label could really be anything since B2C is a white label auth system that can be branded in any way.

**Comment by theyo** [comment]
@jmprieur Just ran into this issue today. Having a default predefined name like "Microsoft Identity Platform" would be great to keep it unique compared to the basic OIDC provider. However, it is also beneficial to be able to change the name in cases where we might need to register mutliple instances of the Microsoft Identity provider as in the case of using something like Identity Server to provide a single federated auth system for a suite of applications.

**Comment by jmprieur** [comment]
@jennyf19. I propose we take this one as well for next release.

Spec.

1. In this line, we need to pass-in a new string parameter the identityProviderName (AddOpenIdConnect method has overrides)

https://github.com/AzureAD/microsoft-identity-web/blob/95d9696161db68bd68b84ec32903968c54840498/src/Microsoft.Identity.Web/WebAppExtensions/MicrosoftIdentityWebAppAuthenticationBuilderExtensions.cs#L275

2. To pass-in this parameter, we cannot do it through the MicrosoftIdentityOptions as they are not yet available at that time (they will be on the call back to the creation of the OpenIdConnect scheme), so I believe we need to pass it as an additional optional parameters to all the AddMicrosoftIdentityWebApp methods

```CSharp
string identityProviderName = "Microsoft identity platform"
```

**Comment by theyo** [comment]
I would also ask that the implementation is consistent with the the built-in providers here: https://github.com/dotnet/aspnetcore/tree/2670c128d522473e146ff9f8159bfffdfe694cd9/src/Security/Authentication

Namely, that the parameter name `displayName` and possibly the parameter order in the extension methods and the addition of a `MicrosoftIdentityWebDefaults` class.

**Comment by jmprieur** [comment]
@theyo : what is the rationale for this ask? the displayName makes sense for AddOpenIdConnect, but AddMicrosoftIdentityWebApp does way more than that. 
Would we agree with `identityProviderDisplayName` ?

**Comment by theyo** [comment]
@jmprieur I don't think the comparison is between `AddMicrosoftIdentityWebApp` and `AddOpenIdConnect`. It should really be compared to one of the other wrappers like `AddMicrosoftAccount`. I agree that `AddMicrosoftIdentityWebApp` does a lot, but at the end of the day it's a wrapper around `AddOpenIdConnect` just in the same manner that `AddMicrosoftAccount` is a wrapper around `AddOAth`.

This method from `AddMicrosoftAccount` shows how `displayName` is being passed down into the wrapped `AddOAuth` method:
https://github.com/dotnet/aspnetcore/blob/2670c128d522473e146ff9f8159bfffdfe694cd9/src/Security/Authentication/MicrosoftAccount/src/MicrosoftAccountExtensions.cs#L67 

Each of the other built-in ASP.NET providers (Google, Twitter, Facebook) expose `displayName` and pass it into the specific provider they are wrapping in the same manner and this has been the pattern the community typically follows for other provider implementations. Therefore, I believe it makes sense to follow the pattern for the `AddMicrosoftIdentityWebApp` implemention as well.

**Comment by apondman** [comment]
there is a workaround you can use right now until the displayName parameter is more easily available:

`services.PostConfigure<AuthenticationOptions>(options =>
            {
                foreach (var scheme in options.Schemes)
                {
                    if (scheme.Name == OpenIdConnectDefaults.AuthenticationScheme)
                    {
                        scheme.DisplayName = "My Desired Display Name";
                    }
                }
            });`

or register the scheme with a different name:

`.AddMicrosoftIdentityWebApp(section, "MyCustomAuthScheme");`

and replace the line in the first code snippet above with:

`if (scheme.Name == "MyCustomAuthScheme")`

**Comment by jmprieur** [comment]
@tratcher FYI we still need to do that.

**Comment by jennyf19** [comment]
I have a branch: https://github.com/AzureAD/microsoft-identity-web/tree/jennyf/displayName 
with the changes to handle providing the DisplayName, if you want to try it out. Still a work in progress. 
@Shazwazza @apondman @theyo @farshid3003

**Comment by jennyf19** [comment]
Included in [1.16 release](https://github.com/AzureAD/microsoft-identity-web/releases/tag/1.16.0)
